unit_tests/src/test_file_object.cpp

```diff
@@ -0,0 +1,16 @@
+#include <catch2/catch.hpp>
+
+#include "irods/irods_file_object.hpp"
+
+TEST_CASE("file_object")
+{
+    SECTION("test file object size can be negative issue 7154")
+    {
+        irods::file_object obj{};
+        REQUIRE(obj.size() == 0);
+
+        obj.size(-1);
+        REQUIRE(obj.size() == -1);
```

Comment:
Perhaps we should modify this or add another assertion that `obj.size() < 0` to make sure that there's no funny business with casting...?